replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ScheduledReplicatorDeleter.java

```diff
@@ -0,0 +1,248 @@
+
+  private final Security security;
+
+  public ScheduledReplicatorDeleter(
+      ReplicatorConfigManager replicatorConfigManager,
+      ScheduledExecutorService scheduledExecutorService,
+      ReplicationItemManager replicationItemManager,
+      ReplicatorHistory replicatorHistory,
+      Metacards metacards) {
+    this(
+        replicatorConfigManager,
+        scheduledExecutorService,
+        replicationItemManager,
+        replicatorHistory,
+        metacards,
+        Security.getInstance(),
+        TimeUnit.MINUTES.toMillis(1));
+  }
+
+  public ScheduledReplicatorDeleter(
+      ReplicatorConfigManager replicatorConfigManager,
+      ScheduledExecutorService scheduledExecutorService,
+      ReplicationItemManager replicationItemManager,
+      ReplicatorHistory replicatorHistory,
+      Metacards metacards,
+      Security security,
+      long pollPeriod) {
+    this.replicatorConfigManager = replicatorConfigManager;
+    this.scheduledExecutorService = scheduledExecutorService;
+    this.replicationItemManager = replicationItemManager;
+    this.replicatorHistory = replicatorHistory;
+    this.metacards = metacards;
+    this.security = security;
+
+    LOGGER.info(
+        "Scheduling replicator config cleanup every {} seconds",
+        TimeUnit.MILLISECONDS.toSeconds(pollPeriod));
+    scheduledExecutorService.scheduleAtFixedRate(
+        this::cleanup, 0, pollPeriod, TimeUnit.MILLISECONDS);
+  }
+
+  public void destroy() {
+    scheduledExecutorService.shutdownNow();
+  }
+
+  public void cleanup() {
+    security.runAsAdmin(
+        () -> {
+          try {
+            security.runWithSubjectOrElevate(
+                () -> {
+                  List<ReplicatorConfig> replicatorConfigs =
+                      replicatorConfigManager.objects().collect(Collectors.toList());
+
+                  cleanupOrphanedReplicationItems(replicatorConfigs);
+                  cleanupDeletedConfigs(replicatorConfigs);
+
+                  return null;
+                });
+          } catch (SecurityServiceException | InvocationTargetException e) {
+            LOGGER.debug("Failed scheduled cleanup of deleted replicator configs", e);
+          }
+
+          return null;
+        });
+  }
+
+  /**
+   * Deletes {@link ReplicationItem}s which have no corresponding data store entry or {@link
+   * ReplicatorConfig}. This occurs when a {@link ReplicatorConfig} was deleted without cleaning up
+   * data, but the data was then deleted manually afterwards.
+   */
+  private void cleanupOrphanedReplicationItems(List<ReplicatorConfig> replicatorConfigs) {
+    Set<String> replicatorConfigIds =
+        replicatorConfigs.stream().map(ReplicatorConfig::getId).collect(Collectors.toSet());
+
+    int startIndex = 0;
+    Set<String> configlessItemIds = Collections.emptySet();
+
+    do {
+      try {
+
+        configlessItemIds =
+            replicationItemManager
+                .getItemsForConfig("", startIndex, DEFAULT_PAGE_SIZE)
```

Comment:
❓ Are we guaranteed that the start index stays consistent from one query to the next? What if new items appears in the mean time? Wouldn't that change the order and all?